x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/exceptions/exceptions_overflow_display.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState } from 'react';
+import { EuiPopover, EuiBadgeGroup, EuiButton } from '@elastic/eui';
+import styled from 'styled-components';
+import { History } from 'history';
+import * as i18n from '../../translations';
+import { Spacer } from '../../../../../../common/components/page';
+import { LinkAnchor } from '../../../../../../common/components/links';
+import { getRuleDetailsUrl } from '../../../../../../common/components/link_to/redirect_to_detection_engine';
+import { Rule } from '../../../../../containers/detection_engine/rules';
+import { FormatUrl } from '../../../../../../common/components/link_to';
+
+interface ExceptionOverflowDisplayProps {
+  rules: Rule[];
+  history: History;
+  formatUrl: FormatUrl;
+}
+
+const ExceptionOverflowWrapper = styled(EuiBadgeGroup)`
+  width: 100%;
+`;
+
+const ExceptionOverflowPopoverWrapper = styled(EuiBadgeGroup)`
+  max-height: 200px;
+  max-width: 600px;
+  overflow: auto;
+`;
+
+const ExceptionOverflowPopoverButton = styled(EuiButton)`
+  font-size: ${({ theme }) => theme.eui.euiFontSizeXS}
+  font-weight: 500;
+  height: 20px;
+`;
+
+/**
+ * @param rules to display for filtering
+ */
+const ExceptionOverflowDisplayComponent = ({
+  rules,
+  history,
+  formatUrl,
+}: ExceptionOverflowDisplayProps) => {
+  const [isExceptionOverflowPopoverOpen, setIsExceptionOverflowPopoverOpen] = useState(false);
+
+  return (
+    <>
+      {rules.length <= 2 ? (
+        <ExceptionOverflowWrapper data-test-subj="rules">
+          {rules.map(({ name, id }, index: number) => (
+            <Spacer key={id}>
+              <LinkAnchor
+                key={id}
+                data-test-subj="ruleName"
+                onClick={(ev: { preventDefault: () => void }) => {
+                  ev.preventDefault();
+                  history.push(getRuleDetailsUrl(id));
+                }}
+                href={formatUrl(getRuleDetailsUrl(id))}
+              >
+                {name}
+              </LinkAnchor>
+              {index !== rules.length - 1 ? ', ' : ''}
+            </Spacer>
```

Comment:
Planning on abstracting this out since it's used 3 times here

Reply:
Sounds good -- third ©️🍝 is a charm. Would you be willing to open an issue for tracking this tech debt?